calculate_outliers_by_threshold.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env pypy
+
+import argparse
+import bz2
+import json
+import math
+import os
+import sys
+
+
+FILENAME = 'outliers_per_threshold.json.bz2'
+WINDOWS = [25, 50, 100, 200, 300, 400]
+
+
+def sum_outliers(data):
+    num_outliers = 0
+    for outliers in data:
+        num_outliers += len(outliers)
+    return num_outliers
+
+
+def get_outliers(all_outliers, window_size, threshold=1):
+    common, unique = list(), list()
+    for index, outliers in enumerate(all_outliers):
+        common_exec = list()
+        unique_exec = list()
+        for outlier in outliers:
+            other_execs = all_outliers[:index] + all_outliers[(index + 1):]
+            sum_ = 0
+            for execution in other_execs:
+                if outlier in execution:
+                    sum_ += 1
+            if sum_ >= threshold:
+                common_exec.append(outlier)
+            else:
+                unique_exec.append(outlier)
+        common.append(common_exec)
+        unique.append(unique_exec)
+    return common, unique
+
+
+def median(data):
+    """Naive algorithm to compute the median of a list of (sorted) data.
+    Linear interpolation is used when the percentile lies between two data
+    points. It is assumed that the data contains no NaN or similar.
+    """
+    size = len(data)
+    if size == 0:
+        raise ValueError('Cannot compute percentile of empty list!')
+    if size == 1:
+        return data[0]
+    index = (size - 1) // 2
+    if size % 2 == 1:
+        return data[index]
+    else:
+        return (data[index] + data[index + 1]) / 2.0
+
```

Comment:
`median` is a lot faster to run (shaves about 17% off the total execution time vs. using `percentile`).